File: python/spec/support/file_to_list.py
#!/usr/bin/env python3
#

def file_to_list(filename):
  lyst = []

  with open(filename, 'r') as file:
    for lyne in file:
      lyne = lyne.strip()
      if len(lyne):
        lyst.append(lyne)

  return lyst

File: go/support/file_to_list.go
package aoc2025_support

import (
	"bufio"
	"os"
	"strings"
)

func FileToList(filename string) []string {
	lyst := make([]string, 0)

	file, oops := os.Open(filename)
	if (oops == nil) {
		defer file.Close()

		scanner := bufio.NewScanner(file)
		for scanner.Scan() {
			text := strings.TrimSpace(scanner.Text())
			if len(text) > 0 {
				lyst = append(lyst, text)
			}
		}
	}

	return lyst
}

